lib/new_relic/agent/instrumentation/net_http/instrumentation.rb

```diff
@@ -32,12 +32,27 @@ def request_with_tracing(request)
             end
 
             wrapped_response = NewRelic::Agent::HTTPClients::NetHTTPResponse.new(response)
```

Comment:
This is the POC for adding headers from the response; it's broken without the LLM events PR atm.